src/Word.java

```diff
@@ -1,78 +1,63 @@
+    guessedChars = new HashSet<>();
+  }
 
-            if (unguessableChars.contains(c)) {
-                return GuessResult.UNGUESSABLE_CHAR;
-            }
+  public int wordLength(){
+    return (int) originalWord.chars().mapToObj(i -> (char) i).filter(Character::isLetter).count();
+  }
 
-            if (guessedChars.contains(c)) {
-                return GuessResult.CHAR_ALREADY_GUESSED;
-            }
 
-            guessedChars.add(c);
-
-            if (originalWord.contains(c + "")) {
-                Set<Character> allChars = new HashSet<>(guessedChars);
-                allChars.addAll(unguessableChars);
+  public GuessResult guess(String guess) {
+    if (guess.length() == 0) {
+      return GuessResult.UNGUESSABLE_CHAR;
+    }
+    if (guess.length() == 1) {
+      char c = Character.toLowerCase(guess.charAt(0));
 
-                for (int i = 0; i < originalWord.length(); i++) {
-                    if (!allChars.contains(originalWord.charAt(i))) {
-                        return GuessResult.CORRECT_GUESS;
-                    }
-                }
+      if (!Character.isLetter(c)) {
+        return GuessResult.UNGUESSABLE_CHAR;
+      }
 
-                return GuessResult.COMPLETED_WORD;
-            }
+      if (guessedChars.contains(c)) {
+        return GuessResult.CHAR_ALREADY_GUESSED;
+      }
 
-            return GuessResult.INCORRECT_GUESS;
-        }
+      guessedChars.add(c);
 
-        if (originalWord.equals(guess)) {
-            return GuessResult.COMPLETED_WORD;
+      if (originalWord.contains(c + "") || originalWord.contains(Character.toUpperCase(c) + "")) {
+        if (guessedChars.containsAll(originalWord.chars().mapToObj(i -> Character.toLowerCase((char) i))
+            .filter(Character::isLetter).collect(Collectors.toSet()))){
+          return GuessResult.COMPLETED_WORD;
         }
-
-        return GuessResult.INCORRECT_GUESS;
+        return GuessResult.CORRECT_GUESS;
+      }
     }
 
-    @Override
-    public String toString() {
-        StringBuilder builder = new StringBuilder();
-
-        for (int i = 0; i < originalWord.length(); i++) {
-            char c = originalWord.charAt(i);
-
-            if (unguessableChars.contains(c) ||
-                guessedChars.contains(c)) {
-                builder.append(c);
-            } else {
-                builder.append("_");
-            }
+    if (originalWord.toLowerCase().equals(guess.toLowerCase())) {
+      guessedChars.addAll(guess.toLowerCase().chars()
+          .mapToObj(i->(char)i).collect(Collectors.toList()));
+      return GuessResult.COMPLETED_WORD;
+    }
 
-        }
+    return GuessResult.INCORRECT_GUESS;
+  }
 
-        return builder.toString();
-    }
+  @Override
+  public String toString() {
+    StringBuilder builder = new StringBuilder();
+    originalWord.chars().mapToObj(i -> (char) i).map(c -> !Character.isLetter(c)
+        || guessedChars.contains(Character.toLowerCase(c)) ? c : '_').forEach(builder::append);
```

Comment:
Why do we need to lowercase all of these characters? We should instead ensure that originalWord only contains lowercase characters to begin with, at the construction of the object. That avoid you having to call toLowerCase in a bunch of operations.